File: 231119/선두를 지켜라 3/keep-the-lead-3.py
MAX_t=1000000

N,M=map(int,input().split())

pos_A,pos_B=[0]*(MAX_t+1),[0]*(MAX_t+1)


cur_A=1
for _ in range(N):
    v,t=map(int,input().split())
    for _ in range(t):
        pos_A[cur_A]=pos_A[cur_A-1]+v
        cur_A+=1

cur_B=1
for _ in range(M):
    v,t=map(int,input().split())
    for _ in range(t):
        pos_B[cur_B]=pos_B[cur_B-1]+v
        cur_B+=1

ans=0
leader=-1
for t in range(1,cur_A):
    if pos_A[t]>pos_B[t]:
        ans+=(1 if leader!=1 else 0)
        leader=1
    elif pos_A[t]<pos_B[t]:
        ans+=(1 if leader!=2 else 0)
        leader=2
    else:
        ans+=(1 if leader!=0 else 0)
        leader=0

print(ans)

File: 251209/선두를 지켜라 3/keep-the-lead-3.js
const MAX_T = 1000000;

const fs= require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [N,M] = input[0].split(' ').map(Number);

const posA = Array(MAX_T+1).fill(0);
let timeA = 1;
for(let i=1; i<=N; i++){
    let [v, t] = input[i].split(' ').map(Number);
    for(let j=0; j<t; j++){
        posA[timeA]=posA[timeA-1]+v;
        timeA++;
    }
}

const posB = Array(MAX_T+1).fill(0);
let timeB = 1;
for(let i=N+1; i<input.length; i++){
    let [v, t] = input[i].split(' ').map(Number);
    for(let j=0; j<t; j++){
        posB[timeB]=posB[timeB-1]+v;
        timeB++;
    }
}

if(timeA<timeB){
    for(let i=timeA; i<timeB; i++){
        posA[i]=posA[i-1];
    }
}else if(timeA>timeB){
    for(let i=timeB; i<timeA; i++){
        posB[i]=posB[i-1];
    }
}


let winner =0;
let cnt=0;
for(let i=1; i<Math.max(timeA,timeB); i++){
    if(posA[i]>posB[i]){
        if(winner!==1){
            winner=1;
            cnt++;
        }
    }else if(posA[i]<posB[i]){
        if(winner!==2){
            winner=2;
            cnt++;
        }
    }else{
        if(winner!==3){
            winner=3;
            cnt++;
        }
    }
}

console.log(cnt);